tests/testthat/test-table_parts.R

```diff
@@ -187,6 +187,90 @@ test_that("a gt table contains the expected spanner column labels", {
   )
 })
 
+test_that("`tab_spanner()` exclusively uses IDs for arranging spanners", {
+
+  tbl_html_1 <-
+    dplyr::tibble(A_X = c(1), B_X = c(2), A_Y = c(3), B_Y = c(4)) %>%
+    gt() %>%
+    tab_spanner(label = "A", id = "a", columns = ends_with("X"), gather = FALSE) %>%
+    tab_spanner(label = "A", id = "b", columns = ends_with("Y"), gather = FALSE) %>%
+    tab_style(
+      style = cell_fill("green"),
+      locations = cells_column_spanners(spanners = "a")
+    ) %>%
+    render_as_html() %>%
+    xml2::read_html()
```

Comment:
Let's repeat this example (spanners with same label, different ids) for latex and rtf as well

Reply:
Now added as tests.